File: 4-Trees/java/src/Main.java
import BinarySearchTree.*;

public class Main {

	public static void main(String[] args) {

		BstTree bstTree = new BstTree(10);
		Node root = bstTree.getRoot();
		bstTree.addNode(root, 23);
		bstTree.addNode(root, 42);
		bstTree.addNode(root, 43);
		bstTree.addNode(root, 30);
		bstTree.addNode(root, 2);
		bstTree.addNode(root, 40);

		System.out.println("max = " + bstTree.max());
		System.out.println("min = " + bstTree.min());
		System.out.println("height = " + bstTree.height());

		bstTree.levelOrder();

		bstTree.search(root, 23);
		bstTree.search(root, 10);
		bstTree.search(root, 2341);

	}
}

File: 4-Trees/cpp/main.cpp
#include "bstTree.h"
#include <iostream>

using namespace std;

int main (int argc, char *argv[])
{
	BstTree * bstTree = new BstTree(10);
	Node *root = bstTree->getRoot();

	bstTree->addNode(23, root);
	bstTree->addNode(342, root);
	bstTree->addNode(900, root);
	bstTree->addNode(934, root);
	bstTree->addNode(256, root);
	bstTree->addNode(2, root);
	bstTree->addNode(1004, root);
	bstTree->addNode(23114, root);

	cout << "max = " << bstTree->max() << endl;
	cout << "min = " << bstTree->min() << endl;
	cout << "height = " << bstTree->height() << endl;

	bstTree->levelOrder();

	bstTree->search(23, root);
	bstTree->search(342, root);
	bstTree->search(900, root);

	delete bstTree;
	return 0;
}
